signature/policy_eval_signedby.go

```diff
@@ -70,6 +71,13 @@ func (pr *prSignedBy) isSignatureAuthorAccepted(image types.UnparsedImage, sig [
 			return PolicyRequirementError(fmt.Sprintf("Signature by key %s is not accepted", keyIdentity))
 		},
 		validateSignedDockerReference: func(ref string) error {
+			r := image.Reference().DockerReference()
+			if _, isCanonical := r.(reference.Canonical); isCanonical || reference.IsNameOnly(r) {
+				if !pr.UntaggedReference.matchesDockerReference(image, ref) {
```

Comment:
which leads to:
```
{
   // …
   "referenceByDigest": "reject|acceptableSignedIdentity",
   "signedIdentity": {
      "type": "exactRepository",
      "dockerRepository": "r.a.rh.com/rhel7/rhel",
   }
}
```
so when `reject` -> do not accept canonical references unless the reference in the signature itself is the same image@digest

when `acceptableSignedIdentity` -> accept canonical references from users iff user's sha256 == digest stored in the signature for any tag in the signature's reference.

all set?